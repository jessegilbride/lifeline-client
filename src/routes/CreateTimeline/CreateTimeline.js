import React, { Component, Fragment } from 'react';
import TimelineContext from '../../contexts/TimelineContext';
import TimelineApiService from '../../services/timeline-api-service';
import './CreateTimeline.css';

export default class CreateTimeline extends Component {
  static contextType = TimelineContext;

  handleSubmitTimeline = (event) => {
    event.preventDefault();
    const { line_name, description } = event.target;
    const line =  {
      line_name: line_name.value,
      description: description.value
    };
    TimelineApiService.postTimeline(line)
      .then((res) => {
        // console.log(res)
        return this.context.setTimeline
      })
      .then(() => {
        this.props.history.push('/timeline')
      })
      // .catch(this.context.setError); // use local state
  };

  render() {
    return (
      <Fragment>
        <section>
          <h2 className='page-heading'>Create Timeline</h2>
          <p>(message to user)</p>
        </section>
        <section>
          <form className='create-timeline-form' onSubmit={this.handleSubmitTimeline}>
            <div className='form-section'>
              <label htmlFor='line_name'>Timeline Name</label>
              <input
                type='text'
                id='line_name'
                name='line_name'
                placeholder=''
                required
              />
            </div>
            <div className='form-section'>
              <label htmlFor='description'>Description</label>
              <textarea id='description' name='description' rows='10' required></textarea>
            </div>
            <div className='button-section'>
              <button type='submit'>Submit</button>
            </div>
          </form>
        </section>
      </Fragment>
    );
  }
}
