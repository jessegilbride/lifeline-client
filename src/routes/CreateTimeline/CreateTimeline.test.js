import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
import CreateTimeline from './CreateTimeline';

describe('CreateTimeline component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <MemoryRouter><CreateTimeline /></MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <MemoryRouter><CreateTimeline name="CreateTimeline" /></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});
