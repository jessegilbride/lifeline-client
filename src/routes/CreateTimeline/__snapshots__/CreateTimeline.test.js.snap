// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTimeline component renders the UI as expected 1`] = `
Array [
  <section>
    <h2
      className="page-heading"
    >
      Create Timeline
    </h2>
    <p>
      (message to user)
    </p>
  </section>,
  <section>
    <form
      className="create-timeline-form"
      onSubmit={[Function]}
    >
      <div
        className="form-section"
      >
        <label
          htmlFor="line_name"
        >
          Timeline Name
        </label>
        <input
          id="line_name"
          name="line_name"
          placeholder=""
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          id="description"
          name="description"
          required={true}
          rows="10"
        />
      </div>
      <div
        className="button-section"
      >
        <button
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </section>,
]
`;
